backend/app/api/src/endpoints/organization/dashboard/organization/PatchOrganizationEndpoint.test.ts

```diff
@@ -59,4 +66,1245 @@ describe('Endpoint.PatchOrganization', () => {
+                expect(response.body.privateMeta).not.toBeNull();
+
+                // roles
+                expect(response.body.privateMeta!.roles).toHaveLength(2);
+                expect(response.body.privateMeta!.roles[1].id).toBe(role1.id);
+                expect(response.body.privateMeta!.roles[1].resources.size).toBe(1);
+
+                // level should not be default level
+                expect(response.body.privateMeta!.roles[1].level).toBe(PermissionLevel.None);
+                expect(response.body.privateMeta!.roles[1].accessRights).toBeEmpty();
+            });
+
+            test('should only be able to patch resources of roles', async () => {
+                // arrange
+                const role1 = PermissionRoleDetailed.create({
+                    name: 'role1',
+                });
+                organization.privateMeta.roles = [role1, ...organization.privateMeta.roles];
+
+                await organization.save();
+
+                const rolesPatch: PatchableArrayAutoEncoder<PermissionRoleDetailed> = new PatchableArray();
+
+                rolesPatch.addPatch(PermissionRoleDetailed.patch({
+                    id: role1.id,
+                    name: 'new name',
+                    resources: new PatchMap([
+                        [PermissionsResourceType.Webshops,
+                            new Map([
+                                [webshop.id,
+                                    ResourcePermissions.create({ level: PermissionLevel.Full }),
+                                ],
+                            ]),
+                        ],
+                    ]),
+                    level: PermissionLevel.Full,
+                }));
+
+                const patch = OrganizationStruct.patch({
+                    id: organization.id,
+                    privateMeta: OrganizationPrivateMetaData.patch({
+                        roles: rolesPatch,
+                    }),
+                });
+
+                // act
+                const response = await patchOrganization({
+                    patch,
+                    organization,
+                    token,
+                });
+
+                // assert
+                expect(response.body).toBeDefined();
+                expect(response.body.privateMeta).not.toBeNull();
+
+                // roles
+                expect(response.body.privateMeta!.roles).toHaveLength(2);
+                expect(response.body.privateMeta!.roles[0].id).toBe(role1.id);
+                expect(response.body.privateMeta!.roles[0].resources.size).toBe(1);
+
+                // name should not have been changed
+                expect(response.body.privateMeta!.roles[0].name).toBe('role1');
+                // permission level should not have been changed
+                expect(response.body.privateMeta!.roles[0].level).toBe(PermissionLevel.None);
+            });
+
+            test('should be able to put limited responsibilities', async () => {
```

Comment:
See other comments - putting responsibilities is not allowed so test should test for error instead of success now